### PR TITLE
refactor: cache with context

### PR DESCRIPTION
**Pull request by iwpnd**
to merge after #1012

### CODE REVIEW

File: cache/azblob/azblob_test.go
Comment (gdey) [inline cache/azblob/azblob_test.go:10]: nit this should be with the other standard libraries.

**Review by gdey — APPROVED** [review]
LGTM, just a small nit about import lines.

I need to look at the CI issues as well. Looks like it's an issue with valun, and maybe how vendoring is done?

**Comment by iwpnd** [comment]
> LGTM, just a small nit about import lines.
> 
> I need to look at the CI issues as well. Looks like it's an issue with valun, and maybe how vendoring is done?

that'll be fine after #1012

**Review by ARolek — COMMENTED** [review]
Thank you for tackling this! Tegola is much better off with this change. I have a couple minor requests inline, but they're not show stoppers.

File: cache/azblob/azblob.go
Comment (ARolek) [reply cache/azblob/azblob.go:127]: I just walked through the code and the cache init routine is a bit confusing. The part I'm referencing enhancing with the context is the InitFunc:

https://github.com/go-spatial/tegola/blob/93b917e04ee4a9efb8721bae9b1d2b9ba46c5456/cache/cache.go#L136

This is called by the `For` method:

https://github.com/go-spatial/tegola/blob/93b917e04ee4a9efb8721bae9b1d2b9ba46c5456/cache/cache.go#L165

Which is invoked by the internal `cmd/internal/register/` package:

https://github.com/go-spatial/tegola/blob/93b917e04ee4a9efb8721bae9b1d2b9ba46c5456/cmd/internal/register/caches.go#L16

The main value we would get with this plumbing is if tegola is started up and then killed before the cache initializes fully. 

I'm not going to block the PR on this request, but just following up on the intent of my comment. Your call if you want to make this change ;-)

Comment (iwpnd) [reply cache/azblob/azblob.go:127]: gotcha! if thats fine with you i'd merge this as is, and tackle this in a separate commit.

**Comment by ARolek** [comment]
@iwpnd can you rebase against master and fix the conflicts please?

**Pull request merged**
